jobs/build-mechanical.Jenkinsfile

```diff
@@ -19,28 +19,30 @@ properties([
 
 node {
     def mechanical_streams = pipeutils.streams_of_type(pipecfg, 'mechanical')
+    def scheduled_streams = pipeutils.scheduled_streams(pipecfg, mechanical_streams)
 
-    change = checkout(
-        [$class: 'GitSCM',
-         userRemoteConfigs: [[url: pipecfg.source_config.url]],
-         branches: pipeutils.streams_as_branches(mechanical_streams)
-        ]
-    )
-
-    if (pipeutils.triggered_by_push()) {
-        stream = pipeutils.stream_from_branch(change.GIT_BRANCH)
-        if (stream in mechanical_streams) {
-            build job: 'build', wait: false, parameters: [
-              string(name: 'STREAM', value: stream)
-            ]
-        }
-    } else {
-        // cron or manual build: build all mechanical streams
-        mechanical_streams.each{
+    if (scheduled_streams) {
+        // if any mechanical stream has the `scheduled: true` knob set, only run those streams
+        scheduled_streams.each{
+            echo "Triggering build for scheduled stream: ${it}"
             build job: 'build', wait: false, parameters: [
               string(name: 'STREAM', value: it),
               booleanParam(name: 'EARLY_ARCH_JOBS', value: false)
             ]
         }
+    } else {
+        // only trigger all mechanical FCOS streams. There is another mechanism to trigger
+        // these streams for RHCOS. Since the brew section of the config is required for
+        // RHCOS, use it to decide which product we're building.
+        def is_rhcos = pipecfg.brew != null
+        if (!is_rhcos) {
```

Comment:
Hmm, there's no need for this. The proposed logic should work fine for both FCOS and RHCOS. In general, the pipeline shouldn't know about RHCOS vs FCOS.

Reply:
I think if there are no `scheduled` streams in RHCOS, we can just disable the job. It's already the case today that if somehow `build-mechanical` wasn't dormant (i.e. we ran it at least once so that it sets up its timer), it would trigger builds for every stream.

Reply:
Yeah, we are just likely to forget about it, though it should be pretty obvious when we have 10 builds all starting at the same time. I'll remove the check here and instead I'll add a comment to the rhcos pipecfg to disable the job if the knob is removed.